tests/test_requests.py

```diff
@@ -549,6 +549,11 @@ def test_proxy_error_on_bad_url(self, httpbin, httpbin_secure):
         with pytest.raises(InvalidProxyURL):
             requests.get(httpbin(), proxies={'http': 'http:///example.com:8080'})
 
+    def test_proxy_strip_spaces(self):
+        # Don't yet have testing proxies sorted out.
+        with pytest.raises(ProxyError):
+            requests.get('http://localhost:1', proxies={'http': 'http://localhost:1234 '})
```

Comment:
Would you mind adding a couple more tests for proxies without schemes, leading whitespace, and proxy values of `None`? You should be able to find examples of using pytest's `parametrize` decorator above to do all the tests in one function.

Reply:
- [ ] missing scheme
- [ ] leading whitespace
- [ ] proxy value None